### PR TITLE
feat: Add email scheduling 

### PR DESCRIPTION
**Pull request by kewynakshlley**
- Create email with schedule
- Update email schedule
- Cancel schedule
- Removed date types

### CODE REVIEW

File: src/main/java/com/resend/services/emails/Emails.java [hunk 1]
```diff
@@ -62,7 +58,48 @@ public Email get(String emailId) throws ResendException {
 
             String responseBody = response.getBody();
 
-            Email email = resendMapper.readValue(responseBody, Email.class);
-            return email;
+        return resendMapper.readValue(responseBody, Email.class);
+    }
+
+    /**
+     * Update the schedule of an email by its unique identifier.
```
Comment (felipevolpone) [inline src/main/java/com/resend/services/emails/Emails.java]:
```suggestion
     * Update the email by its unique identifier.
```

In the future this method might also update other attributes

File: src/main/java/com/resend/services/emails/Emails.java [hunk 2]
```diff
@@ -62,7 +58,48 @@ public Email get(String emailId) throws ResendException {
...
+            throw new RuntimeException("Failed to update email: " + response.getCode() + " " + response.getBody());
+        }
+
+        String responseBody = response.getBody();
+
+        return resendMapper.readValue(responseBody, UpdateEmailResponse.class);
+    }
+
+    /**
+     * Cancel the schedule of an email by its unique identifier.
```
Comment (felipevolpone) [inline src/main/java/com/resend/services/emails/Emails.java]:
```suggestion
     * Cancels an email by its unique identifier.
```

File: src/main/java/com/resend/services/emails/model/UpdateEmailOptions.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public static class Builder {
+        private String scheduledAt;
+
+        /**
+         * Set the schedule of the email.
+         *
+         * @param scheduledAt The schedule of the email in ISO 8601 format (e.g., 2024-08-05T11:52:01.858Z).
+         * @return This builder instance for method chaining.
+         */
+        public Builder scheduledAt(String scheduledAt) {
```
Comment (felipevolpone) [inline src/main/java/com/resend/services/emails/model/UpdateEmailOptions.java:49]: Should we instead use `withScheduledAt` in the Builder? If this is how it's done in the others Builders, I vote to keep consistent, but if this is the first, I think that using the `with` prefix makes more sense with the Builder way

Comment (kewynakshlley) [reply src/main/java/com/resend/services/emails/model/UpdateEmailOptions.java:49]: I don't use the 'with' prefix in the other builders. I would prefer to keep it this way without the 'with' prefix to keep the code consistent.

**Review by felipevolpone — APPROVED** [review]

**Pull request merged**